components/eventing-controller/pkg/backend/utils/eventmesh_utils.go

```diff
@@ -156,29 +142,80 @@ func ConvertKymaSubToEventMeshSub(subscription *eventingv1alpha2.Subscription, t
-	//	}
-	//	if subscription.Spec.ProtocolSettings.WebhookAuth.Type == string(types.AuthTypeClientCredentials) {
-	//		auth.Type = types.AuthTypeClientCredentials
-	//	} else {
-	//		return nil, fmt.Errorf("invalid Type: %v", subscription.Spec.ProtocolSettings.WebhookAuth.Type)
-	//	}
-	//	auth.TokenURL = subscription.Spec.ProtocolSettings.WebhookAuth.TokenURL
-	//}
-	eventMeshSubscription.WebhookAuth = auth
+	// set webhook auth
+	eventMeshSubscription.WebhookAuth, err = getEventMeshWebhookAuth(subscription, defaultWebhookAuth)
+	if err != nil {
+		return nil, err
+	}
+
 	return eventMeshSubscription, nil
 }
 
+func setEventMeshProtocolSettings(subscription *eventingv1alpha2.Subscription, eventMeshSub *types.Subscription) error {
+	// Applying protocol settings if provided in subscription CR
+	// content mode
+	if contentMode, ok := subscription.Spec.Config[eventingv1alpha2.ProtocolSettingsContentMode]; ok && contentMode != "" {
+		eventMeshSub.ContentMode = contentMode
+	}
+	// qos
+	if qosStr, ok := subscription.Spec.Config[eventingv1alpha2.ProtocolSettingsQos]; ok && qosStr != "" {
+		qos, err := getQos(qosStr)
+		if err != nil {
+			return err
+		}
+		eventMeshSub.Qos = qos
+	}
+	// handshake
+	if exemptHandshake, ok := subscription.Spec.Config[eventingv1alpha2.ProtocolSettingsExemptHandshake]; ok {
+		handshake, err := strconv.ParseBool(exemptHandshake)
+		if err != nil {
+			handshake = true
+		}
+		eventMeshSub.ExemptHandshake = handshake
+	}
+	return nil
+}
+
+// getEventMeshWebhookAuth uses default webhook auth unless specified in Subscription CR.
+func getEventMeshWebhookAuth(subscription *eventingv1alpha2.Subscription,
+	defaultWebhookAuth *types.WebhookAuth) (*types.WebhookAuth, error) {
+	auth := &types.WebhookAuth{}
+	// extract auth info from subscription CR if any
+	if authType, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthType]; ok {
+		if authType != string(types.AuthTypeClientCredentials) {
+			return nil, fmt.Errorf("invalid Type: %v, required: %v", authType,
+				string(types.AuthTypeClientCredentials))
+		}
+		auth.Type = types.AuthTypeClientCredentials
+	}
+
+	if tokenURL, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthTokenURL]; ok {
+		auth.TokenURL = tokenURL
+	}
+
+	if clientID, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthClientID]; ok {
+		auth.ClientID = clientID
+	}
+
+	if clientSecret, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthClientSecret]; ok {
+		auth.ClientSecret = clientSecret
+	}
+
+	if grantType, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthGrantType]; ok {
+		if grantType != string(types.GrantTypeClientCredentials) {
+			return nil, fmt.Errorf("invalid GrantType: %v, required: %v", grantType,
+				string(types.GrantTypeClientCredentials))
+		}
+		auth.GrantType = types.GrantTypeClientCredentials
+	}
+
+	// check if auth was provided in subscription CR
+	if auth.Type != "" {
+		return auth, nil
+	}
```

Comment:
I think it is better to do the check in the beginning of the function instead of the end:

```golang
if auth.Type == "" {
    return defaultWebhookAuth, nil
}
```

Reply:
As you like, but the current implementation already knows the answer to line 213 `if auth.Type != ""` when the line 189 `auth.Type = types.AuthTypeClientCredentials` is evaluated, then does some checks/computations for the (TokenURL, ClientID and ClientSecret) then if it might throw away those checks/computations if `auth.Type == "" is true`.

Reply:
valid point, I updated